### PR TITLE
New source: Zenhub Graphql

### PR DESCRIPTION
**Pull request by Jgerardopine**
## What
* Adding a new source: Zenhub

## How
* This solution gets all the issues form the zenhub repositories by providing a namespace and (optional) repositories 

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
*Are there any breaking changes? What is the end result perceived by the user?*
No Breaking changes

*For connector PRs, use this section to explain which type of semantic versioning bump occurs as a result of the changes. Refer to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines for more information. **Breaking changes to connectors must be documented by an Airbyte engineer (PR author, or reviewer for community PRs) by using the [Breaking Change Release Playbook](https://docs.google.com/document/d/1VYQggHbL_PN0dDDu7rCyzBLGRtX-R3cpwXaY8QxEgzw/edit).***

*If there are breaking changes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.*

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- Connector version is set to `0.0.1`
    - `Dockerfile` has version `0.0.1`
- Documentation updated
    - Connector's `README.md`
    - Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - `docs/integrations/README.md`

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Connector Generator</strong></summary>

- Issue acceptance criteria met
- PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/resources/pull-requests-handbook)
- If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:generateScaffolds` then checking in your changes
- Documentation which references the generator is updated as needed

</details>

<details><summary><strong>Updating the Python CDK</strong></summary>

### Airbyter

Before merging:
- Pull Request description explains what problem it is solving
- Code change is unit tested
- Build and my-py check pass
- Smoke test the change on at least one affected connector
   - On Github: Run [this workflow](https://github.com/airbytehq/airbyte/actions/workflows/connectors_tests.yml), passing `--use-local-cdk --name=source-<connector>` as options
   - Locally: `airbyte-ci connectors --use-local-cdk --name=source-<connector> test`
- PR is reviewed and approved
      
After merging:
- [Publish the CDK](https://github.com/airbytehq/airbyte/actions/workflows/publish-cdk-command-manually.yml)
   - The CDK does not follow proper semantic versioning. Choose minor if this the change has significant user impact or is a breaking change. Choose patch otherwise.
   - Write a thoughtful changelog message so we know what was updated.
- Merge the platform PR that was auto-created for updating the Connector Builder's CDK version
   - This step is optional if the change does not affect the connector builder or declarative connectors.

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-zenhub-graphql/source_zenhub_graphql/source.py [hunk 1]
```diff
@@ -37,19 +37,18 @@
...
-        #ws_repo_ids = ZenhubWorkspace(api_key=config["access_token"]
-        #    , workspace_name=config["workspace_name"]
-        #    ) 
-        #response, status_code = ws_query_body.fetch_data()
-
-        #return response
-    
-    @property
-    def http_method(self) -> str:
-        return "POST"
```
Comment (Jgerardopine) [inline airbyte-integrations/connectors/source-zenhub-graphql/source_zenhub_graphql/source.py]: I put the POST property in the wrong class.

File: airbyte-integrations/connectors/source-zenhub-graphql/source_zenhub_graphql/streams.py [hunk 2]
```diff
@@ -61,12 +61,13 @@ class ZenhubGraphqlStream(HttpStream, ABC):
     #    self.headers = {'Authorization': f'Bearer {api_key}'}
     #    super().__init__(url_base, self.headers, **kwargs) 
 
-     
+    @property
+    def http_method(self) -> str:
+        return "POST"
```
Comment (Jgerardopine) [inline airbyte-integrations/connectors/source-zenhub-graphql/source_zenhub_graphql/streams.py:66]: This is the fix

File: airbyte-integrations/connectors/source-zenhub-graphql/README.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
+# Zenhub Graphql Source
```
Comment (girarda) [inline airbyte-integrations/connectors/source-zenhub-graphql/README.md:1]: why do we need a new source? Can't we update the existing zenhub connector?

Comment (Jgerardopine) [reply airbyte-integrations/connectors/source-zenhub-graphql/README.md:1]: Yes, you are right. It is not relevant for the user. We will call it Zenhub only

File: airbyte-integrations/connectors/source-zenhub-graphql/integration_tests/sample_config.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+  "access_token": "<user's_access_token>"
+  , "workspace_name": "<your_workspace>"
+  , "repo_names": ["yout_list","of_repos"]
```
Comment (bleonard) [inline airbyte-integrations/connectors/source-zenhub-graphql/integration_tests/sample_config.json:4]: typos

File: airbyte-integrations/connectors/source-zenhub-graphql/README.md (hunk 3 above)
Comment (bleonard) [reply airbyte-integrations/connectors/source-zenhub-graphql/README.md:1]: +1

File: airbyte-integrations/connectors/source-zenhub-graphql/source_zenhub_graphql/streams.py [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+
+        if 'errors' in response_json:
+            raise Exception(f"Errors in response: {response_json['errors']}")
+        
+        issues = response_json.get('data', {}).get('searchIssues', {}).get('nodes', [])
+        print("issues", issues[0])
+        for issue in issues:
+            pipeline_issue= issue.get('pipelineIssue', {}) or {}
+            pipeline = pipeline_issue.get('pipeline', {}) or {}
+            prioirity = pipeline_issue.get('priority', {}) or {}
```
Comment (bleonard) [inline airbyte-integrations/connectors/source-zenhub-graphql/source_zenhub_graphql/streams.py:294]: typo in variable name

File: airbyte-integrations/connectors/source-zenhub-graphql/source_zenhub_graphql/streams.py [hunk 6]
```diff
@@ -0,0 +1,346 @@
...
+            filters={
+                'repositoryIds': self.repo_ids 
+                , 'pipelineIds': self.pipeline_ids
+
+            }
+        )
+        issues_query.totalCount()
+        issue_node = issues_query.nodes.__as__(Issue)
+        issue_node.id()
+        issue_node.ghId()
```
Comment (bleonard) [inline airbyte-integrations/connectors/source-zenhub-graphql/source_zenhub_graphql/streams.py:258]: This builds the query? Interesting.

File: airbyte-integrations/connectors/source-zenhub-graphql/integration_tests/configured_catalog.json [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+        "supported_sync_modes": ["full_refresh", "incremental"]
+      },
+      "sync_mode": "full_refresh",
+      "destination_sync_mode": "overwrite"
+    },
+    {
+      "stream": {
+        "name": "zenhub_issues",
+        "json_schema": {},
+        "supported_sync_modes": ["full_refresh", "incremental"]
```
Comment (bleonard) [inline airbyte-integrations/connectors/source-zenhub-graphql/integration_tests/configured_catalog.json:25]: Do you know how you are going to do incremental syncs here?
I see the `searchIssuesByPipeline` query in the [Explorer](https://developers.zenhub.com/explorer). It looks like it has a sort order. Maybe by `udpated_at`. Is that the plan?

Comment (Jgerardopine) [reply airbyte-integrations/connectors/source-zenhub-graphql/integration_tests/configured_catalog.json:25]: Yes, I am thinking on using updated_at

File: airbyte-integrations/connectors/source-zenhub-graphql/source_zenhub_graphql/streams.py (hunk 6 above)
Comment (Jgerardopine) [reply airbyte-integrations/connectors/source-zenhub-graphql/source_zenhub_graphql/streams.py:258]: Correct

**Comment by natikgadzhi** [comment]
We can get back to this sometime, but for now — closing.

**Pull request closed without merging**